example/randomStrategies/src/randomStrategies.cpp

```diff
@@ -0,0 +1,220 @@
+        , workdivResult{alpaka::getValidWorkDiv<Acc>(
+              alpaka::getDevByIdx<Acc>(0u),
+              extentResult,
+              Vec(static_cast<Idx>(NUM_ROLLS)), // One thread per "point"; each performs NUM_ROLLS "rolls"
+              false,
+              alpaka::GridBlockExtentSubDivRestrictions::Unrestricted)}
+        , bufHostResult{alpaka::allocBuf<float, Idx>(alpaka::getDevByIdx<Host>(0u), extentResult)}
+        , bufAccResult{alpaka::allocBuf<float, Idx>(alpaka::getDevByIdx<Acc>(0u), extentResult)}
+    {
+    }
+};
+
+/// Save the results file in CSV format. One line for each "point"; line length is the number of "rolls".
+void saveToFile(std::string filename, float const* buffer, Box const& box)
+{
+    std::ofstream output;
+    output.open(filename, std::ios::out);
+    std::cout << "Writing " << filename << std::endl;
+    auto lineLength = box.extentResult[0] / box.extentRand[0];
+    for(Box::Idx i = 0; i < box.extentResult[0]; ++i)
+    {
+        output << buffer[i] << ((i + 1) % lineLength ? ", " : "\n");
+    }
+    output.close();
+}
+
+template<Strategy TStrategy>
+void runStrategy(Box& box)
+{
+    // Set up the pointer to the PRNG states buffer
+    alpaka::rand::Philox4x32x10<Box::Acc>* const ptrBufAccRand{alpaka::getPtrNative(box.bufAccRand)};
+
+    // Initialize the PRNG and its states on the device
+    InitRandomKernel<TStrategy> initRandomKernel;
+    if constexpr(TStrategy == Strategy::offset) // offset strategy needs an additional parameter for
+                                                // initialisation: the offset cannot be deduced form the size of
+                                                // the PRNG buffer and has to be passed in explicitly
+    {
+        alpaka::exec<Box::Acc>(
+            box.queue,
+            box.workdivRand,
+            initRandomKernel,
+            box.extentRand,
+            ptrBufAccRand,
+            box.extentResult[0] / box.extentRand[0]); // == NUM_ROLLS; amount of work to be performed by each thread
+    }
+    else // other strategies deduce the initial parameters solely from the thread index
+    {
+        alpaka::exec<Box::Acc>(box.queue, box.workdivRand, initRandomKernel, box.extentRand, ptrBufAccRand);
+    }
+    alpaka::wait(box.queue);
+
+    // OPTIONAL: copy the the initial states to host if you want to check them yourself
+    // alpaka_rand::Philox4x32x10<Box::Acc>* const ptrBufHostRand{alpaka::getPtrNative(box.bufHostRand)};
+    // alpaka::memcpy(box.queue, box.bufHostRand, box.bufAccRand, box.extentRand);
+    // alpaka::wait(box.queue);
+
+    // Set up the pointers to the results buffers
+    float* const ptrBufHostResult{alpaka::getPtrNative(box.bufHostResult)};
+    float* const ptrBufAccResult{alpaka::getPtrNative(box.bufAccResult)};
+
+    // Initialise the results buffer to zero
+    for(Box::Idx i = 0; i < box.extentResult[0]; ++i)
+        ptrBufHostResult[i] = 0;
```

Comment:
memset

Reply:
I still don't like losing the type and treating `ptrBufHostResult` as a byte array. If someone copies the example with memset but tries to initialize to non-zero, they'd be in for a nasty surprise.